src/hooks/useUploadsManager/useUploadsManager.tsx

```diff
@@ -1,62 +1,187 @@
+import {
+  InputAssetUpload,
+  AssetUploadWithProgress,
+  UploadManagerValue,
+  UploadsProgressRecord,
+  StartFileUploadOptions,
+} from './types'
 import { createStorageNodeUrl } from '@/utils/asset'
+import { LiaisonJudgement } from '@/api/queries'
+
+const RETRIES_COUNT = 5
+const RECONNECTION_ERROR_MESSAGE = 'Reconnection failed'
+
+type GroupByParentObjectIdAcc = {
+  [key: string]: AssetUploadWithProgress[]
+}
+
+type AssetFile = {
+  contentId: string
+  blob: File | Blob
+}
 
 const UploadManagerContext = React.createContext<UploadManagerValue | undefined>(undefined)
 UploadManagerContext.displayName = 'UploadManagerContext'
 
 export const UploadManagerProvider: React.FC = ({ children }) => {
+  const navigate = useNavigate()
   const { uploadsState, addAsset, updateAsset } = useUploadsManagerStore()
+  const { displaySnackbar } = useSnackbar()
   const [uploadsProgress, setUploadsProgress] = useState<UploadsProgressRecord>({})
+  const [assetsFiles, setAssetsFiles] = useState<AssetFile[]>([])
+  const { activeUser } = useActiveUser()
+  const channelId = activeUser.channelId ?? ''
+  const { channel, loading: channelLoading } = useChannel(channelId)
+  const { videos, loading: videosLoading } = useVideos(
+    {
+      where: {
+        id_in: uploadsState.filter((item) => item.parentObject.type === 'video').map((item) => item.parentObject.id),
+      },
+    },
+    { skip: !uploadsState.length }
+  )
+
+  const uploadsStateWithProgress: AssetUploadWithProgress[] = uploadsState.map((asset) => ({
+    ...asset,
+    progress: uploadsProgress[asset.contentId] ?? 0,
+  }))
+
+  const channelDataObjects = [channel?.avatarPhotoDataObject, channel?.coverPhotoDataObject]
+  const videosDataObjects = videos?.flatMap((video) => [video.mediaDataObject, video.thumbnailPhotoDataObject]) || []
+  const allDataObjects = [...channelDataObjects, ...videosDataObjects]
+
+  // Enriching data with pending/accepted/rejected status
+  const uploadsStateWithLiaisonJudgement = uploadsStateWithProgress.map((asset) => {
+    const dataObject = allDataObjects.find((dataObject) => dataObject?.joystreamContentId === asset.contentId)
+
+    return { ...asset, liaisonJudgement: dataObject?.liaisonJudgement, ipfsContentId: dataObject?.ipfsContentId }
+  })
+
+  const lostConnectionAssets = uploadsStateWithLiaisonJudgement.filter(
+    (asset) => asset.liaisonJudgement === LiaisonJudgement.Pending
+  )
+
+  useEffect(() => {
+    if (!lostConnectionAssets.length) {
+      return
+    }
+    displaySnackbar({
+      title: `(${lostConnectionAssets.length}) Asset${
+        lostConnectionAssets.length > 1 ? 's' : ''
+      } waiting to resume upload`,
+      description: 'Reconnect files to fix the issue',
+      actionText: 'See assets',
+      onActionClick: () => navigate(absoluteRoutes.studio.uploads()),
+      iconType: 'warning',
+    })
+  }, [lostConnectionAssets.length])
```

Comment:
If we cannot do anything with it  let's just add above ` // eslint-disable-next-line react-hooks/exhaustive-deps`